### ISSUE TITLE
<Resource options={{ label: 'resource.myResource.myKey' }} /> does not get translated

### ISSUE THREAD

**Issue by kopax** [marmelab/react-admin#4547]
I have tried the following:

    <Resource
        options={{ label: 'resource.myResource.myKey' }}
    />

According to documentation: https://marmelab.com/react-admin/Resource.html

> `options.label` allows to customize the display name of a given resource in the menu.

and https://marmelab.com/react-admin/Translation.html

> ```js
> // don't do this
> <TextField source="first_name" label={translate('myroot.first_name')} />
> 
> // do this instead
> <TextField source="first_name" label="myroot.first_name" />
> ```

I don't know if it's a bug but it looks like it.

React-admin 3.3.0

**Comment by fzaninotto** [comment]
Where doesn't it get translated? In the menu? According to the react-admin code, it seems that it should - or perhaps react-admin is showing the key because you didn't provide a translation. 

I suggest you add a CodeSandbox demonstrating the issue to help us reproduce it.

**Comment by JulienMattiussi** [comment]
I found someting.
If your language file contains a `resources.myResource.name` entry, then this is always used and your `options={{ label: 'something' }}` isn't used
But when you remove `resources.myResource.name` from your language file, then the open.label you setted is used and correctly translated.

Does it fit with your case @kopax ?

**Comment by fzaninotto** [comment]
That's right, there are 2 options to customize the label. If you use both, one will be ignored. We should probably throw an exception in that case so that the problem isn't considered as a bug.

**Comment by dkocich** [comment]
- I specified `label` on my resource but it won't translate. I can see that it only always changes depending on the text of a key but it won't change to the real translation key (still uses `customer.list` as a label). Might it be caused because `const translate = useTranslate();` is used while Admin's `i18nProvider` is not yet known?

- I tried to put my `<Resource>` components inside admin and encapsulate it in `<React.Fragment>` but nothing was rendered in the sidebar (I also tried to force rerender but with no luck) - however, I could see that their translations are logged correctly from `getLabel` function ('translated customer.list to Customer List') 

- did anyone find any solution since the last comments?

here are my minimal components
```
import React, { Fragment, useEffect, useState } from "react";
import { Admin, Resource, useTranslate } from "react-admin";
import { authProvider } from "./dataProviders/authProvider";
import dataProvider from "./dataProviders/dataProvider";
import { i18nProvider } from "./i18n/i18nProvider";

function App(): JSX.Element {
  const translate = useTranslate();

  function getLabel(key: string) {
    console.log(`translated ${key} to ${translate(key)}`);
    return translate(key);
  }
  return (
    <div className="App">
      <Admin
        dashboard={Dashboard}
        authProvider={authProvider}
        dataProvider={dataProvider}
        i18nProvider={i18nProvider}
        layout={FleetLayout}
      >
        <Resource
          name="customer"
          options={{
            label: getLabel("customer.list"),
          }}
      </Admin>
    </div>
  );
}
export default App;
```


```
function MyChildResourcesCmp() {
  const [, forceUpdate] = useState();

  useEffect(() => {
    console.log("forcing");
    setTimeout(forceUpdate, 2000);
  }, []);

  const translate = useTranslate();

  function getLabel(key: string) {
    console.log(`translated ${key} to ${translate(key)}`);
    return translate(key);
  }
  return (
    <>
      <Resource
        name="customer"
        list={CustomerList}
        edit={CustomerEdit}
        show={CustomerShow}
        options={{
          label: getLabel("customer.list"),
        }}
      />
    </>
  );
}
```

**Comment by dkocich** [comment]
@kopax were you able to get it working?

**Comment by Jerem33320** [comment]
Hello, it seems that you could use i18nProvider.translate() directly to avoid this problem. Not sure if it is the best way to do it but it works to use translate in resources. For the rest of your components you still can use useTranslate() :)
